nix-ts-mode.el

```diff
@@ -193,14 +221,30 @@
     (setq-local treesit-font-lock-feature-list
                 '((comment builtin)
                   (keyword string path)
-                  (number constant attribute)
+                  (number constant attribute error)
                   (bracket delimiter operator ellipses function)))
 
+    ;; Comments
+    (setq-local comment-start "# ")
+    (setq-local comment-start-skip "#+\\s-*")
+
     ;; Indentation
+    (setq-local indent-tabs-mode nil)
     (setq-local treesit-simple-indent-rules nix-ts-mode-indent-rules)
 
+    ;; Imenu.
+    (setq-local treesit-simple-imenu-settings
+                `(("Bind" "\\`binding\\'" nil nil)))
+
+    ;; Navigation.
+    (setq-local treesit-defun-type-regexp (rx (or "binding")))
+    (setq-local treesit-defun-name-function #'nix-ts-mode--defun-name)
+
     (treesit-major-mode-setup)))
 
+(if (treesit-ready-p 'nix)
+    (add-to-list 'auto-mode-alist '("\\.nix\\'" . nix-ts-mode)))
```

Comment:
I always wonder how useful this clause is: it only adds the auto-mode-alist entry if this file is loaded, so for it to have any effect, people have to preload the file manually. That's why I didn't add it myself.

(Also I'd say `when` is preferable to `if` in the case that there's no "else" form.)

Reply:
I noticed that most non-core-Emacs `-ts-mode` packages use an `;;;###autoload`-ed clause like this for `auto-mode-alist`, which is both more aggressive and kinda makes sense. I have frequently said to MELPA submitters that "package installation does not imply package activation", but a common exception is `auto-mode-alist` entries for _regular_ major modes. In the case of modes that require treesitter, it all just gets more complicated — if I install this mode and start an Emacs 29 without the ocaml treesitter grammar installed, I can no longer edit OCaml. I didn't review the other `-ts-mode` packages in MELPA, but I suspect I would have pointed this out.

Reply:
Thinkng to myself, I wonder how it would be to use something like this:
```el
;;;###autoload
(when (treesit-ready-p 'nix)
  (add-to-list 'auto-mode-alist '("\\.nix\\'" . nix-ts-mode))
```

Reply:
```suggestion
```